packages/nimble-components/src/rich-text/specs/mention-hld.md

```diff
@@ -0,0 +1,721 @@
+    3. When a user repositions the cursor between the text segments added after the mention character.
+
+    Refer the [accessibility](#accessibility) section to know more details about when it is required to emit the event for performing the filtering in the client application.
+
+    Conversely, this event will not be fired when:
+
+    1. A user moves the cursor away either using the mouse or any key down events that move the cursor from the current position to outside of `@`
+       and characters after a minimum of two white spaces while the popup is open.
+    2. A user selects an option from the dropdown list either using mouse click/pressing Enter or Tab.
+    3. A user adds two or more spaces after triggering the mention popup.
+    4. In the absence of `nimble-rich-text-mention-users` and no mapping elements.
+
+_Methods_
+
+-   `checkValidity()` - this returns `true` if the configuration of the `@mention` mapping data is valid and `false` otherwise.
+
+#### User mapping element (Non-visible configuration element):
+
+This mapping element is employed to establish a connection between the value displayed in the mapping view and the corresponding value stored within
+the markdown string. For instance, the username for an `@mention` is contained in the `display-name` attribute, which is used for display in the
+mention the view, while the user ID URL is contained in the `mention-url` attribute which is used to store within the markdown string.
+
+_Component Name_
+
+-   `nimble-mapping-mention-user`
+
+_Props/Attrs_
+
+-   `mention-url`: string
+-   `display-name`: string
+
+#### User mention view (Visible UI element):
+
+This is a UI component used to render the `@mention` node in a rich text editor and rich text viewer when parsed or added into the components.
+This holds the styling for the `@mention` nodes.
+
+Different copying and pasting behaviors of the user mention view node:
+
+1.  Copying an `@mention` name from the viewer and pasting it into the editor should result in the appearance of the same mention node.
+2.  When copying an `@mention` name from the editor and pasting it into the editor should result in the appearance of the same mention node.
+3.  If a portion of an `@mention` name from the viewer is copied and pasted into the editor, it should still render as a complete mention node with the entire name.
+4.  Copying only a portion of an `@mention` name from the editor is not possible, as the entire name will always be selected. The copy-and-paste behavior will be the same as point 2.
+
+_Component Name_
+
+-   `nimble-rich-text-user-mention-view`
+
+_Props/Attrs_
+
+-   `mention-url`: string - the unique user URL containing a user ID of the mentioned user
+-   `mention-label`: string - the user name of the mentioned user or the user ID extracted from the above URL if mapping element is not
+    present for the particular user
+-   `type`: string - the type of the mentioned node, _defaults_ as `mention`
```

Comment:
Since this is added to the default setting of the mention node in Tiptap, we thought that it is required to be added for our custom mention view node. But we quickly tried today with the [prototype](https://github.com/ni/nimble/pull/1593/) branch and seems it is not a required attribute except for stylings. Also, we have a custom node that encapsulates all stylings within an element, we can remove this attribute and have only `mention-href` and `display-name` for `getMarkdown()`.

I removed the `type` from the HLD.